src/RegisterContext.js

```diff
@@ -211,8 +211,7 @@ RegisterContext.prototype = {
     options = options || {};
 
     if(!this.registered && !options.all) {
-      this.logger.warn('already unregistered');
-      return;
```

Comment:
It is in the test cleanup and how we are cleaning up the UA that is causing the tests to break. We are blindly calling unregister in places, and expecting it to return out of the function if it is not registered.

The issue is, that it is valid to send an unregister even if you are not registered. There are cases where you would want to do this. Either we need to re-write the tests, and confirm that we are not breaking anything. Or we need to keep the current behavior and add some extra behavior to allow this to continue in certain situations.